zebra-state/src/sled_state.rs

```diff
@@ -153,6 +153,27 @@ impl FinalizedState {
                 .block
                 .coinbase_height()
                 .expect("valid blocks must have a height");
+
+            if prev_hash == block::Hash([0; 32]) {
+                assert_eq!(
+                    1,
+                    self.block_by_height.len(),
+                    "cannot commit genesis block to a populated state"
+                );
+                assert_eq!(
+                    block::Height(0),
+                    height,
+                    "cannot commit genesis: invalid height"
+                );
+            } else {
+                assert_eq!(
+                    self.finalized_tip_height()
+                        .expect("state must have a genesis block committed")
+                        + 1,
+                    Some(height)
+                );
+            }
```

Comment:
I think these checks might be easier to understand if they're written in terms of the state, rather than the content of the block we're about to commit.
```suggestion
            if self.block_by_height.is_empty() {
                assert_eq!(
                    block::Hash([0; 32]),
                    prev_hash,
                    "the first block added to an empty state must be a genesis block"
                );
                assert_eq!(
                    block::Height(0),
                    height,
                    "cannot commit genesis: invalid height"
                );
            } else {
                assert_eq!(
                    self.finalized_tip_height()
                        .expect("state must have a genesis block committed")
                        + 1,
                    Some(height)
                );
            }
```